src/FunctionApps/python/GenerateCSVs/fhir.py

```diff
@@ -12,17 +14,49 @@
 RECORD_TYPE_ECR = "ecr"
 
 
-def read_bundles_by_type() -> Iterator[Tuple[str, dict]]:
+def get_blobs(
+    container_url: str, container_prefix: str, record_types: list
+) -> Iterator[Tuple[str, IO]]:
     """
     Read FHIR bundles from blob storage and return an iterator
     along with the source of the record (one of the RECORD_TYPE_ constants)
     """
+    client = get_container_client(container_url)
+    for record_type in record_types:
+        # record_type is upper-cased in the blob server
+        for props in client.list_blobs(
+            name_starts_with=container_prefix + record_type.upper()
+        ):
+            logging.info(f"reading blob {props.name}")
+            if props.size > 0:
+                # If it's an actual file, download it and yield out the
+                # individual records along with record type
+                blob_client = client.get_blob_client(props)
+                yield (
+                    record_type,
+                    io.BytesIO(blob_client.download_blob().content_as_bytes()),
+                )
 
-    return
+
+def read_bundles_by_type(
+    container_url: str, container_prefix: str, record_types: list
+) -> Iterator[Tuple[str, dict]]:
+    """Reads FHIR bundle dicts and returns an iterator referencing tuples containing
+    the following information:
+    record_type (str) The record type (e.g. ELR, ECR, VXU)
```

Comment:
You may want to mention that the values would be defined in the constants (and I think they're lowercase)

Reply:
Updated.